server/sonar-process-monitor/src/main/java/org/sonar/application/process/ProcessLauncherImpl.java

```diff
@@ -61,17 +63,34 @@ public void close() {
   }
 
   @Override
+  public ProcessMonitor launch(EsCommand esCommand) {
+    Process process = null;
+    try {
+      ProcessBuilder processBuilder = create(esCommand);
+      LOG.info("Launch process[{}]: {}", esCommand.getProcessId().getKey(), String.join(" ", processBuilder.command()));
+
+      process = processBuilder.start();
+
+      return new EsProcessMonitor(process, esCommand);
+    } catch (Exception e) {
+      // just in case
+      if (process != null) {
+        process.destroyForcibly();
+      }
+      throw new IllegalStateException(format("Fail to launch process [%s]", esCommand.getProcessId().getKey()), e);
+    }
+  }
+
+  @Override
   public ProcessMonitor launch(JavaCommand javaCommand) {
     Process process = null;
-    ProcessCommands commands;
     try {
-      commands = allProcessesCommands.createAfterClean(javaCommand.getProcessId().getIpcIndex());
+      ProcessCommands commands = allProcessesCommands.createAfterClean(javaCommand.getProcessId().getIpcIndex());
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "ProcessCommands" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2095)